solutions/cf-1557B.mdx

```diff
@@ -0,0 +1,345 @@
+			<th>2</th>
+			<th>3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th className="bg-green-100 dark:bg-green-800">2</th>
+			<th>3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th className="bg-green-100 dark:bg-green-800">2</th>
+			<th className="bg-red-100 dark:bg-red-800">3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th className="bg-green-100 dark:bg-green-800">2</th>
+			<th className="bg-red-100 dark:bg-red-800">3</th>
+			<th className="bg-blue-100 dark:bg-blue-800">4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+Thus, we need 4 subarrays at minimum.
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+<Spoiler title="Time Complexity Proof">
+Sorting Requires $\mathcal{O}(N\log N)$, and repeated map insertions also require $\mathcal{O}(N\log N)$. Hence the overall time complexity is $\mathcal{O}(N\log N)$.
+</Spoiler>
+	
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+typedef long long ll;
+#define pb push_back
+#define FF first
+#define SS second
+ 
+void solve() {
+	int n, k; cin >> n >> k;
+	int arr[n];
+	int sarr[n];
+	for (int i = 0; i < n; i++) {
+		cin >> arr[i];
+		sarr[i] = arr[i];
+	}
+	map<int, int> next; // Map from one value to the value after it in the sorted array
+	sort(sarr, sarr + n);
+	for (int i = 0; i < n - 1; i++) {
+		next[sarr[i]]=sarr[i+1];
+	}
+	next[sarr[n-1]] = 1000000007; // Ridiculously Large Number
+	int ans = 1;
+	for (int i = 1; i < n; i++) {
+		// If the previous element in the unsorted array is not the same 
+		// as the previous element in the sorted array, we need at least 1 more segment
+		if (next[arr[i-1]]!=arr[i]) ans++; 
+	}
+	cout << (ans <= k ? "YES" : "NO") << endl;
+}
+ 
+int main() {
+	ios::sync_with_stdio(0); cin.tie(0);
+	int t; cin >> t; while (t--) solve();
```

Comment:
statements on one line
also a for loop would be really, really nice

Reply:
while(t--) is incredibly common. If you check any C++ solution with testcases on CF you are most likely going to find a while loop.

Reply:
that still doen't mean it's good

Reply:
I don't see why it shouldn't be used...